src/unity/toolkits/activity_classification/activity_classifier.cpp

```diff
@@ -447,6 +496,19 @@ void activity_classifier::init_train(
   data_params.predictions_in_chunk = NUM_PREDICTIONS_PER_CHUNK;
   training_data_iterator_ = create_iterator(data_params);
 
+  // Bind the validation data to a data iterator.
+  if (variant_is<gl_sframe>(validation_data)) {
+    gl_sframe validation_sf = variant_get_value<gl_sframe>(validation_data);
+    data_iterator::parameters validation_data_params;
+    validation_data_params.data = validation_sf;
+    validation_data_params.target_column_name = target_column_name;
+    validation_data_params.session_id_column_name = session_id_column_name;
+    validation_data_params.feature_column_names = feature_column_names;
+    validation_data_params.prediction_window =
+        read_state<flex_int>("prediction_window");
+    validation_data_params.predictions_in_chunk = NUM_PREDICTIONS_PER_CHUNK;
+    validation_data_iterator_ = create_iterator(validation_data_params);
+  }
```

Comment:
Don't forget to add the else case here, to guard against multiple calls to `init_train`